clippy_lints/src/unit_types/let_unit_value.rs

```diff
@@ -71,25 +71,46 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, local: &'tcx LetStmt<'_>) {
                 local.span,
                 "this let-binding has unit value",
                 |diag| {
+                    let mut suggestions = Vec::new();
+
+                    // Suggest omitting the `let` binding
                     if let Some(expr) = &local.init {
                         let mut app = Applicability::MachineApplicable;
                         let snip = snippet_with_context(cx, expr.span, local.span.ctxt(), "()", &mut app).0;
-                        diag.span_suggestion(local.span, "omit the `let` binding", format!("{snip};"), app);
+                        suggestions.push((local.span, format!("{snip};")));
                     }
 
-                    if let PatKind::Binding(_, binding_hir_id, ident, ..) = local.pat.kind
-                        && let Some(body_id) = cx.enclosing_body.as_ref()
-                        && let body = cx.tcx.hir().body(*body_id)
-                        && is_local_used(cx, body, binding_hir_id)
-                    {
-                        let identifier = ident.as_str();
-                        let mut visitor = UnitVariableCollector::new(binding_hir_id);
-                        walk_body(&mut visitor, body);
-                        visitor.spans.into_iter().for_each(|span| {
-                            let msg =
-                                format!("variable `{identifier}` of type `()` can be replaced with explicit `()`");
-                            diag.span_suggestion(span, msg, "()", Applicability::MachineApplicable);
-                        });
+                    // If this is a binding pattern, we need to add suggestions to remove any usages
+                    // of the variable
+                    if let PatKind::Binding(_, binding_hir_id, ident, ..) = local.pat.kind {
+                        if let Some(body_id) = cx.enclosing_body.as_ref() {
```

Comment:
This should still be a let-chain:

```suggestion
                    if let PatKind::Binding(_, binding_hir_id, ident, ..) = local.pat.kind
                        && let Some(body_id) = cx.enclosing_body.as_ref() {
```

This will change the indentation of all the code below.